src/services/DealService.ts

```diff
@@ -108,7 +110,52 @@ export class DealService {
     /**
      * deals will take care of themselves on account changes
      */
-    this.getDeals();
+    if (!this.deals) {
+      await this.getDeals();
+    }
+
+    this.observeDeals();
+  }
+
+  private async observeDeals(): Promise<void> {
+    this.deals = new Map<Address, DealTokenSwap>();
+
+    const dealsObservable = await this.dataSourceDeals.getDealsObservables(this.ethereumService.defaultAccountAddress);
+
+    if (this.dealsSubscription) {
+      this.dealsSubscription.unsubscribe();
+    }
+
+    this.dealsSubscription = dealsObservable.subscribe(
+      updates => {
+
+        try {
+          if (!updates) {
+            throw new Error("Deals are not accessible");
+          }
+
+          const dealsMap = new Map<Address, DealTokenSwap>();
+
+          for (const dealDoc of updates.removed) {
+            this.deals.delete(dealDoc.id);
+          }
+
+          for (const dealDoc of updates.modified) {
+            this._createDeal(dealDoc, dealsMap);
```

Comment:
If only modified, why are you creating a new entity?  Don't we simply want to replace the  existing deal.dealDocument?